common/setups/hybrid/gmm_system.py

```diff
@@ -0,0 +1,878 @@
+            + meta.align_then_split_and_accumulate_sequence(
+                splits, accs_per_split, mark_align=False
+            )
+            + ["align!"]
+        )
+
+        akv = dict(**self.default_align_keep_values)
+        if align_keep_values is not None:
+            akv.update(align_keep_values)
+
+        self.train(
+            name=name,
+            corpus=corpus,
+            sequence=action_sequence,
+            flow="%s+cmllr" % feature_flow,
+            initial_alignment=meta.select_element(self.alignments, corpus, alignment),
+            align_keep_values=akv,
+            **kwargs,
+        )
+        self.jobs[corpus]["train_{}".format(name)].selected_alignment_jobs[
+            -1
+        ].add_alias("train_{}_align_last".format(name))
+        self.jobs[corpus]["train_{}".format(name)].selected_mixture_jobs[-1].add_alias(
+            "train_{}_mix_last".format(name)
+        )
+        tk.register_output(
+            "train_{}_align_bundle_last".format(name),
+            self.jobs[corpus]["train_{}".format(name)]
+            .selected_alignment_jobs[-1]
+            .alignment_bundle,
+        )
+        tk.register_output(
+            "train_{}_mix_last".format(name),
+            self.jobs[corpus]["train_{}".format(name)]
+            .selected_mixture_jobs[-1]
+            .mixtures,
+        )
+
+    # -------------------- recognition  --------------------
+
+    def recognition(
+        self,
+        name,
+        iters,
+        corpus,
+        feature_flow,
+        feature_scorer,
+        pronunciation_scale,
+        lm_scale,
+        search_params,
+        rtf,
+        mem,
+        parallelize_conversion,
+        lattice_to_ctm_kwargs,
+        **kwargs,
+    ):
+        with tk.block(f"{name}_recognition"):
+            optimize_am_lm_scale = (
+                self.gmm_monophone_args.monophone_recognition_args.pop(
+                    "optimize_am_lm_scale", False
+                )
+            )
+            recog_func = self.recog_and_optimize if optimize_am_lm_scale else self.recog
+
+            for it in iters:
+                recog_func(
+                    name=f"{name}-{it:02d}",
+                    prefix=f"recog_{corpus}_{name}/",
+                    corpus=corpus,
+                    flow=feature_flow,
+                    feature_scorer=list(feature_scorer) + [it - 1],
+                    pronunciation_scale=pronunciation_scale,
+                    lm_scale=lm_scale,
+                    search_parameters=search_params,
+                    rtf=rtf,
+                    mem=mem,
+                    parallelize_conversion=parallelize_conversion,
+                    lattice_to_ctm_kwargs=lattice_to_ctm_kwargs,
+                    **kwargs,
+                )
+
+    # -------------------- run setup  --------------------
+
+    def run(self, steps=tuple("all")):
+        assert len(steps) > 0
+        if len(steps) == 1 and steps[0] == "all":
```

Comment:
This can be simplified:
```suggestion
        if list(steps) == ["all"]:
```
Which I find more readable.

Or:
```suggestion
        if tuple(steps) == ("all",):
```